configure.ac

```diff
@@ -23,6 +23,17 @@ AC_CONFIG_HEADERS([config.h])
 AC_CANONICAL_HOST
 AM_INIT_AUTOMAKE([-Wall std-options subdir-objects])
 
+AC_ARG_ENABLE(
+   [build-as-unsupported],
+   [AS_HELP_STRING(
+      [--enable-build-as-unsupported],
+      [Force build as an unsupported platform])
+   ],
+   [if test "x$enable_build_as_unsupported" = "xyes"; then
+      host_os="unsupported"
```

Comment:
I think you did it wrong. You should use the variable`my_htop_platform`. `host_os` is an Autoconf-reserved variable that might imply a cross-compiler should be used.

Reply:
Just a personal idea: I don't like adding a configure option for something that is meant for developers (porting to new platforms).
I think an easier method is to allow an undocumented override of `my_htop_platform`, and then a builder can run configure like this:

```
export my_htop_platform=unsupported; ./configure --enable-werror --enable-unicode
```

Reply:
Currently there's no easy way to influence `my_htop_platform` without duplicating much of the platform selection logic. And I'd actually also be fine if compiling for the `unsupported` platform was treated/handled as a cross-build.

I think, if we want to keep `host_os` untouched, we could initialize `htop_host_os` as `host_os` and do the override on `htop_host_os`.

I'm open to alternate suggestions. The goal for this PR is mostly to get the code for the `unsupported` platform into the usual CI builds, and using an explicit configure switch seemed like the most straight forward solution to me.

Reply:
@BenBE The problem with `--enable-build-as-unsupported` is that it conflicts with `--enable-pcp` as both options would override `my_htop_platform`.

Reply:
@BenBE The problem with `--enable-build-as-unsupported` is that it conflicts with `--enable-pcp` as both options would override `my_htop_platform`.

Reply:
I also noticed, that `./configure --host=none` seems to work equally well …

Reply:
@BenBE That's perhaps unintentional. You should specify `--host=x86_64-unknown-unsupported_os` instead. That's the "target triplet" format.

Reply:
Nope, that's intentional and correct, as `config.sub` translates single-part information into the full triplet.

Cf. [config.sub](https://git.savannah.gnu.org/cgit/autoconf.git/tree/build-aux/config.sub) for the full implementation …

FWIW:
```console
$ ./configure --host=x86_64-unknown-unsupported_os
checking build system type... x86_64-pc-linux-gnu
checking host system type... Invalid configuration `x86_64-unknown-unsupported_os': OS `unsupported_os' not recognized
configure: error: /bin/bash ./build-aux/config.sub x86_64-unknown-unsupported_os failed
```

A full, arch-independent version could be:
```console
$ ./configure --host=$(uname -m)-unknown-none
```

Reply:
After a bit of thinking, I now believe your approach `./configure --host=$(uname -m)-none` might be the cleanest one. As long as there is no cross-compiler named `$(uname -m)-none-gcc`, `configure` would be happy to grab the native compiler to do what we've intended.